### PR TITLE
Chapter 8

### PR DESCRIPTION
**Pull request by Josesosa07**

### CODE REVIEW

File: estate/views/estate_property_offer_views.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <record id="estate_property_offer_view_form" model="ir.ui.view">
+        <field name="name">estate.property.offer.form</field>
+        <field name="model">estate.property.offer</field>
+        <field name="arch" type="xml">
+            <form>
+                <sheet>
+                    <group>
+                        <field name="price"/>
+                        <field name="partner_id"/>
+                        <field name="state"/>
```
Comment (EmilioSerna) [inline estate/views/estate_property_offer_views.xml:18]: Isn't `property_id` also required by the model?

File: estate/views/estate_property_type_views.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                </sheet>
+            </form>
+        </field>
+    </record>
+    
+    <record id="estate_property_type_view_tree" model="ir.ui.view">
+        <field name="name">estate.property.type.tree</field>
+        <field name="model">estate.property.type</field>
+        <field name="arch" type="xml">
+            <tree string="Property Types">
```
Comment (EmilioSerna) [inline estate/views/estate_property_type_views.xml]:
```suggestion
            <tree>
```
This attribute `string` is deprecated in tree views, can you please remove it?

File: estate/models/estate_property.py [hunk 3]
```diff
@@ -1,48 +1,57 @@
...
 
 
 class EstateProperty(models.Model):
+    # --------------------------------------- Private Attributes ----------------------------------
+
     _name = "estate.property"
     _description = "Real Estate Property"
 
+    # --------------------------------------- Fields Declaration ----------------------------------
     # name = fields.Char('Estate Property Name', required=True, translate=True)
```
Comment (EmilioSerna) [inline estate/models/estate_property.py]:
```suggestion
```
Please remove commented code

File: estate/models/estate_property_offer.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    _name = "estate.property.offer"
+    _description = "Real Estate Property Offer"
+
+    # --------------------------------------- Fields Declaration ----------------------------------
+
+    # Basic
+    price = fields.Float()
+    state = fields.Selection(selection=_status, copy=False, string="Status")
+    partner_id = fields.Many2one('res.partner', string='Partner', required=True)
+    property_id = fields.Many2one('estate.property', string='Property', required=True)
```
Comment (EmilioSerna) [inline estate/models/estate_property_offer.py]:
```suggestion
    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
    property_id = fields.Many2one("estate.property", string="Property", required=True)
```

**Review by EmilioSerna — CHANGES_REQUESTED** [review]
@Josesosa07 These comments are the same from the [previous Pull Request](https://github.com/Josesosa07/training_odoo16/pull/1)

These are the last comments I see, please rebase and edit the commit messages with the convention tags and messages like: [IMP] estate: ...

Also, squash the commits with repeated messages or edit them to be different. Depending if the changes are related or no.

File: estate/views/estate_property_type_views.xml (hunk 2 above)
Comment (Josesosa07) [reply estate/views/estate_property_type_views.xml]: Solved

File: estate/models/estate_property.py (hunk 3 above)
Comment (Josesosa07) [reply estate/models/estate_property.py]: Done

File: estate/models/estate_property_offer.py (hunk 4 above)
Comment (Josesosa07) [reply estate/models/estate_property_offer.py]: Done

File: estate/views/estate_property_offer_views.xml (hunk 1 above)
Comment (Josesosa07) [reply estate/views/estate_property_offer_views.xml:18]: Despite the fact that the property_id field is required, and it is not include it in the views. The documentation states that Odoo knows which property is linked to as part of the magic of using the Odoo framework: sometimes things are defined implicitly. It states that when we create a record through a one2many field, the corresponding many2one is populated automatically for convenience, so I think that it is ok.

Comment (EmilioSerna) [reply estate/views/estate_property_offer_views.xml:18]: You're right!

That's okay

**Review by EmilioSerna — CHANGES_REQUESTED** [review]
@Josesosa07 

All the comments from the code are solved, the only thing left is to edit the messages from all these commits to comply to the conventions.

`git rebase -i {{hash_commit}}`

With the file open of the rebase, choose `reword` instead of `pick`, after that you will going commit after commit editing all the messages

Also, squash the commits with repeated messages or edit them to be different. Depending if the changes are related or no.

**Review by EmilioSerna — APPROVED** [review]

**Pull request merged**